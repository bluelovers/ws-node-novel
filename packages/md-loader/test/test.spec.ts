import { parse } from '../index';

describe(`simple`, () =>
{
	let k = parse(`
---
title: Home
---
# novel

- title: 破滅の魔導王とゴーレムの蛮妃
`, {
		parseOptions: {
			//lowCheckLevel: true,
			throw: false,
		},
	});

	test(`parse`, () =>
	{
		expect(k).toMatchSnapshot();
	});

	test(`k.stringify(k)`, () =>
	{

		let actual = k.stringify(k);
		let expected;

		expect(actual).toMatchSnapshot();

	});

	test(`k.stringify v2`, () =>
	{

		let actual = k.stringify({
			data: {
				//...k.data,
				//kkk: 777,
			},
			//content: k.content,
			mdconf: {
				...k.mdconf,
				aaa: {
					bbb: 777,
				},
			},
		});;
		let expected;

		expect(actual).toMatchSnapshot();

	});

})
