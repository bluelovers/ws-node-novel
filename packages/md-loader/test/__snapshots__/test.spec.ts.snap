// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple k.stringify v2 1`] = `
"# novel

- title: 破滅の魔導王とゴーレムの蛮妃
- author: 
- source: 
- cover: 
- publisher: 
- date: 
- status: 

## series

- name: 

## preface


\`\`\`

\`\`\`

## tags

- node-novel

# contribute


# options

## textlayout


# aaa

- bbb: 777


"
`;

exports[`simple k.stringify(k) 1`] = `
"---
title: Home
---
# novel

- title: 破滅の魔導王とゴーレムの蛮妃
"
`;

exports[`simple parse 1`] = `
Object {
  "content": "# novel

- title: 破滅の魔導王とゴーレムの蛮妃
",
  "data": Object {
    "title": "Home",
  },
  "matter": Object {
    "content": "# novel

- title: 破滅の魔導王とゴーレムの蛮妃
",
    "data": Object {
      "title": "Home",
    },
    "excerpt": "",
    "isEmpty": false,
  },
  "mdconf": Object {
    "novel": Object {
      "title": "破滅の魔導王とゴーレムの蛮妃",
    },
    "options": Object {
      "textlayout": Object {},
    },
  },
  "stringify": [Function],
}
`;
